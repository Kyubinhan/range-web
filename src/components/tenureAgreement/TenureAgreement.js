import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import debounce from 'lodash.debounce';

import TenureAgreementList from './TenureAgreementList';
import TenureAgreementSearch from './TenureAgreementSearch';

import { getMockTenureAgreements } from './test/mockValues';

const propTypes = {
  tenureAgreements: PropTypes.array.isRequired,
  searchTenureAgreements: PropTypes.func.isRequired,
}

const defaultProps = {
  tenureAgreements: getMockTenureAgreements(6),
  searchTenureAgreements: (term) => {
    console.log(term);
  },
}

export class TenureAgreement extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
    // wait for 1 second for the user to finish writing a search term
    // then make a network call
    this.searchTenureAgreements = debounce(props.searchTenureAgreements, 1000);
  }  
  
  handleSearchInput = (searchTerm) => {
    this.searchTenureAgreements(searchTerm)
  }
  
  render() {
    const { tenureAgreements } = this.props;

    return (
      <div className="tenure-agreement">
        <div className="tenure-agreement__header">
          Tenure Agreements
        </div>

        <div className="tenure-agreement__search">
          <TenureAgreementSearch
            label="Search Agreements:"
            placeholder="Enter Search Terms..." 
            handleSearchInput={this.handleSearchInput}
          />
        </div>

        <TenureAgreementList 
          tenureAgreements={tenureAgreements}
        />
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {

  }; 
};

TenureAgreement.propTypes = propTypes;
TenureAgreement.defaultProps = defaultProps;

export default connect(
  mapStateToProps, null
)(TenureAgreement);