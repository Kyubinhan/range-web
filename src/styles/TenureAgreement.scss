.tenure-agreement {
  display: flex;
  flex-direction: column;
  padding: 15px;
  
  &__header {
    font-size: 35px;	
    font-weight: bold;
    line-height: 40px;
  }

  &__search {
    display: flex;
    justify-content: flex-end;
    margin-bottom: 15px;
  }
}

.tenure-agreement-list {
  li:first-child {
    border-top: 4px solid;
  }
  li:nth-child(2n) {
    background: #fafafa;
  }
}

.tenure-agreement-search {
  display: flex;
  align-items: center;

  &__label {
    font-size: 15px;	
    font-weight: bold;	
    line-height: 20px;
    margin-right: 10px;
  }
}

.tenure-agreement-list-item {
  list-style: none;
  border-bottom: 1px solid;

  &__content {
    display: flex;
    height: 50px;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;

    &__left,
    &__right {
      display: flex;
    }
  
    &__number {
      font-size: 1.2rem;
    }
  
    &__info {
      font-size: 1.2rem;
    }
  
    &__divider {
      margin: 0 30px;
      border-right: 1px solid;
    }
  }

  &__collapse {
    display: none;

    &--active {
      display: block;
    }
  }
}
