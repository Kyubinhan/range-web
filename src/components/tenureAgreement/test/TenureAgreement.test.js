import React from 'react';
import { shallow } from 'enzyme';
import { TenureAgreement } from '../TenureAgreement';

const props = {};
const setupProps = () => {
  props.searchTenureAgreements = jest.fn();
  props.tenureAgreements = [],
  props.isLoading = false;
};

beforeEach(() => {
  setupProps();
});

describe('TenureAgreement', () => {
  xit('renders correctly', () => {
    const wrapper = shallow(<TenureAgreement {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders `TenureAgreementSearch and TenureAgreementTable`', () => {
    const wrapper = shallow(<TenureAgreement {...props} />);
    expect(wrapper.find('TenureAgreementSearch').exists()).toBe(true);    
    expect(wrapper.find('TenureAgreementTable').exists()).toBe(true);    
  });

  describe('Event handlers', () => {
    it('`handleSearchInput` calls searchTenureAgreement function', () => {
      const wrapper = shallow(<TenureAgreement {...props} />);
      const instance = wrapper.instance();
      const mockSearchTerm = 'search';
      const searchTenureAgreementSpy = jest.spyOn(instance, 'searchTenureAgreements');
      
      instance.handleSearchInput(mockSearchTerm);
      expect(searchTenureAgreementSpy).toHaveBeenCalledWith(mockSearchTerm);
    });
  });
});