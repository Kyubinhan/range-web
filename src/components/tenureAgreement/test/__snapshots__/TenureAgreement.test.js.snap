// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenureAgreement renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TenureAgreement
    rangeUsePlans={Array []}
    searchTenureAgreements={[Function]}
    tenureAgreements={
        Array [
            Object {
              "number": 1,
              "tenureHolder": Object {
                "name": "tenure holder #1",
              },
            },
            Object {
              "number": 2,
              "tenureHolder": Object {
                "name": "tenure holder #2",
              },
            },
            Object {
              "number": 3,
              "tenureHolder": Object {
                "name": "tenure holder #3",
              },
            },
            Object {
              "number": 4,
              "tenureHolder": Object {
                "name": "tenure holder #4",
              },
            },
            Object {
              "number": 5,
              "tenureHolder": Object {
                "name": "tenure holder #5",
              },
            },
            Object {
              "number": 6,
              "tenureHolder": Object {
                "name": "tenure holder #6",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="tenure-agreement__header"
>
          Tenure Agreements
</div>,
        <div
          className="tenure-agreement__search"
>
          <TenureAgreementSearch
                    handleSearchInput={[Function]}
                    label="Search Agreements:"
                    placeholder="Enter Search Terms..."
          />
</div>,
        <TenureAgreementList
          tenureAgreements={
                    Array [
                              Object {
                                "number": 1,
                                "tenureHolder": Object {
                                  "name": "tenure holder #1",
                                },
                              },
                              Object {
                                "number": 2,
                                "tenureHolder": Object {
                                  "name": "tenure holder #2",
                                },
                              },
                              Object {
                                "number": 3,
                                "tenureHolder": Object {
                                  "name": "tenure holder #3",
                                },
                              },
                              Object {
                                "number": 4,
                                "tenureHolder": Object {
                                  "name": "tenure holder #4",
                                },
                              },
                              Object {
                                "number": 5,
                                "tenureHolder": Object {
                                  "name": "tenure holder #5",
                                },
                              },
                              Object {
                                "number": 6,
                                "tenureHolder": Object {
                                  "name": "tenure holder #6",
                                },
                              },
                            ]
          }
/>,
      ],
      "className": "tenure-agreement",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tenure Agreements",
          "className": "tenure-agreement__header",
        },
        "ref": null,
        "rendered": "Tenure Agreements",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TenureAgreementSearch
            handleSearchInput={[Function]}
            label="Search Agreements:"
            placeholder="Enter Search Terms..."
/>,
          "className": "tenure-agreement__search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSearchInput": [Function],
            "label": "Search Agreements:",
            "placeholder": "Enter Search Terms...",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "tenureAgreements": Array [
            Object {
              "number": 1,
              "tenureHolder": Object {
                "name": "tenure holder #1",
              },
            },
            Object {
              "number": 2,
              "tenureHolder": Object {
                "name": "tenure holder #2",
              },
            },
            Object {
              "number": 3,
              "tenureHolder": Object {
                "name": "tenure holder #3",
              },
            },
            Object {
              "number": 4,
              "tenureHolder": Object {
                "name": "tenure holder #4",
              },
            },
            Object {
              "number": 5,
              "tenureHolder": Object {
                "name": "tenure holder #5",
              },
            },
            Object {
              "number": 6,
              "tenureHolder": Object {
                "name": "tenure holder #6",
              },
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="tenure-agreement__header"
>
            Tenure Agreements
</div>,
          <div
            className="tenure-agreement__search"
>
            <TenureAgreementSearch
                        handleSearchInput={[Function]}
                        label="Search Agreements:"
                        placeholder="Enter Search Terms..."
            />
</div>,
          <TenureAgreementList
            tenureAgreements={
                        Array [
                                    Object {
                                      "number": 1,
                                      "tenureHolder": Object {
                                        "name": "tenure holder #1",
                                      },
                                    },
                                    Object {
                                      "number": 2,
                                      "tenureHolder": Object {
                                        "name": "tenure holder #2",
                                      },
                                    },
                                    Object {
                                      "number": 3,
                                      "tenureHolder": Object {
                                        "name": "tenure holder #3",
                                      },
                                    },
                                    Object {
                                      "number": 4,
                                      "tenureHolder": Object {
                                        "name": "tenure holder #4",
                                      },
                                    },
                                    Object {
                                      "number": 5,
                                      "tenureHolder": Object {
                                        "name": "tenure holder #5",
                                      },
                                    },
                                    Object {
                                      "number": 6,
                                      "tenureHolder": Object {
                                        "name": "tenure holder #6",
                                      },
                                    },
                                  ]
            }
/>,
        ],
        "className": "tenure-agreement",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tenure Agreements",
            "className": "tenure-agreement__header",
          },
          "ref": null,
          "rendered": "Tenure Agreements",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TenureAgreementSearch
              handleSearchInput={[Function]}
              label="Search Agreements:"
              placeholder="Enter Search Terms..."
/>,
            "className": "tenure-agreement__search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleSearchInput": [Function],
              "label": "Search Agreements:",
              "placeholder": "Enter Search Terms...",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "tenureAgreements": Array [
              Object {
                "number": 1,
                "tenureHolder": Object {
                  "name": "tenure holder #1",
                },
              },
              Object {
                "number": 2,
                "tenureHolder": Object {
                  "name": "tenure holder #2",
                },
              },
              Object {
                "number": 3,
                "tenureHolder": Object {
                  "name": "tenure holder #3",
                },
              },
              Object {
                "number": 4,
                "tenureHolder": Object {
                  "name": "tenure holder #4",
                },
              },
              Object {
                "number": 5,
                "tenureHolder": Object {
                  "name": "tenure holder #5",
                },
              },
              Object {
                "number": 6,
                "tenureHolder": Object {
                  "name": "tenure holder #6",
                },
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
